js/ui/discoveryFeedButton.js

```diff
@@ -54,21 +50,42 @@ function maybeCreateButton() {
  */
 const DiscoveryFeedButton = new Lang.Class({
     Name: 'DiscoveryFeedButton',
-    Extends: St.Button,
+    Extends: St.BoxLayout,
 
     _init: function() {
-        this.parent({ name: 'discovery-feed',
-                      child: new St.Icon({ icon_name: 'go-down-symbolic',
-                                           icon_size: 16,
-                                           style_class: 'discovery-feed-icon ' }),
-                      style_class: 'discovery-feed-button',
-                      visible: _primaryMonitorWidthPassesThreshold() });
+        this.parent({ vertical: true,
+                      style_class: 'discovery-feed-button' });
+
+        this._bar = new St.Button({ name: 'discovery-feed-bar',
+                                    child: new St.Icon({ style_class: 'discovery-feed-bar-icon' }),
+                                    style_class: 'discovery-feed-bar',
+                                    visible: _primaryMonitorWidthPassesThreshold() });
+        this.add(this._bar);
+
+        this._tile = new St.Button({ name: 'discovery-feed-tile',
+                                    child: new St.Icon({ style_class: 'discovery-feed-tile-icon' }),
+                                    style_class: 'discovery-feed-tile',
+                                    visible: _primaryMonitorWidthPassesThreshold() });
```

Comment:
Instead of controlling the visibility state of both elements separately with the same function, is it possible to move this visibility check into the parent?